src/Maestro/Microsoft.DotNet.Maestro.Tasks/src/PushMetadataToBuildAssetRegistry.cs

```diff
@@ -295,6 +299,18 @@ private string GetVersion(string assetId)
                 {
                     var manifest = (Manifest)xmlSerializer.Deserialize(stream);
 
+                    if (publishingVersion == -1)
+                    {
+                        publishingVersion = manifest.PublishingVersion;
```

Comment:
Legacy manifests may not have the publishing version (I think master sets it appropriately right now, but 3.x may not have that change). I would omit the attribute altogether in the merged manifest if it doesn't exist in the input manifests.

Reply:
Well if it is not set legacy is assumed which won't create a MergedManifest so there's no need to not include it

Reply:
Also, this same code will work since `manifest.PublishingVersion` will end up being 0 so the same logic would work

Reply:
The publishing infra will treat 0 as an error:

https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.Build.Tasks.Feed/src/PublishArtifactsInManifest.cs#L213-L232

Reply:
Also see: https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.Build.Tasks.Feed/src/PublishArtifactsInManifest.cs#L168-L188